core/src/main/java/org/pentaho/platform/dataaccess/datasource/api/MetadataService.java

```diff
@@ -101,16 +102,15 @@ public void removeMetadata( String metadataId ) throws PentahoAccessControlExcep
 
   public List<String> getMetadataDatasourceIds() {
     List<String> metadataIds = new ArrayList<String>();
-    try {
-      sleep( 100 );
-      for ( String id : metadataDomainRepository.getDomainIds() ) {
-        if ( isMetadataDatasource( id ) ) {
-          metadataIds.add( id );
-        }
+    Set<String> domainIds = metadataDomainRepository.getDomainIds();
+    List<String> domainsIdArr = new ArrayList<>( domainIds );
```

Comment:
Why do this ? You can iterate over the set just fine - no need to copy it to an arraylist